kong/core/router.lua

```diff
@@ -464,7 +464,7 @@ function _M.new(apis)
     method = upper(method)
 
 
-    local host = headers["host"] or headers["Host"]
+    local host = headers["X-Host-Override"] or headers["x-host-override"] or headers["host"] or headers["Host"]
```

Comment:
This is to say, even when the test provides something like `{ ["Host"] = "example.com" }`:

https://github.com/Mashape/kong/blob/master/spec/02-integration/05-proxy/01-router_spec.lua#L266

The headers table derived in `kong/core/router` is always built from `ngx.req.get_headers` where the `raw` parameter is not true; therefore, this conditional is unnecessary.